src/Microsoft.ML.Data/Model/Pfa/PfaUtils.cs

```diff
@@ -180,36 +180,45 @@ private static JToken PfaTypeOrNullCore(ColumnType itemType)
                 {
                     // Keys will retain the property that they are just numbers,
                     // with 0 representing missing.
-                    if (keyType.Count > 0 || itemType.RawKind != DataKind.U8)
+                    if (keyType.Count > 0 || keyType.RawType != typeof(ulong))
                         return Int;
                     return Long;
                 }
 
-                switch (itemType.RawKind)
+                System.Type rawType = itemType.RawType;
+                if (rawType == typeof(sbyte)
+                    || rawType == typeof(byte)
+                    || rawType == typeof(short)
+                    || rawType == typeof(ushort)
+                    || rawType == typeof(int))
                 {
-                    case DataKind.I1:
-                    case DataKind.U1:
-                    case DataKind.I2:
-                    case DataKind.U2:
-                    case DataKind.I4:
-                        return Int;
-                    case DataKind.U4:
-                    case DataKind.I8:
-                    case DataKind.U8:
-                        return Long;
-                    case DataKind.R4:
-                    // REVIEW: This should really be float. But, for the
+                    return Int;
+                }
+                else if (rawType == typeof(uint)
```

Comment:
>uint [](start = 43, length = 4)

not your change, but i wonder if uint belongs here, and not with the clause above, where typeof(int) is included. 
AFAIK int and uint have the same length. Are we going by numeric ranges?

Reply:
I assume if you are using `uint`, you have the possibility of having values that are greater than the signed `int.MaxValue`. Thus, if we did use `Int`, it would overflow to a negative value, which I'm sure wouldn't produce the correct results.

---
In reply to: [249968334](https://github.com/dotnet/machinelearning/pull/2176#discussion_r249968334) [](ancestors = 249968334)